src/frontend/src/catalog/source_catalog.rs

```diff
@@ -28,6 +29,28 @@ pub struct SourceCatalog {
     pub source_type: SourceType,
 }
 
+impl SourceCatalog {
+    /// Extract `field_descs` from `column_desc` and add in source catalog.
+    pub fn rewrite_columns(&mut self) {
+        let mut catalogs = vec![];
+        for col in &self.columns {
+            // Extract `field_descs` and return `column_catalogs`.
+            catalogs.append(
+                &mut col
+                    .column_desc
+                    .get_column_descs()
+                    .into_iter()
+                    .map(|c| ColumnCatalog {
+                        column_desc: c,
+                        is_hidden: col.is_hidden,
+                    })
+                    .collect_vec(),
+            )
+        }
+        self.columns = catalogs.clone();
+    }
```

Comment:
```suggestion
    pub fn flatten(mut self) -> Self {
        self.columns.iter().map(|col| {
                col
                    .column_desc
                    .flatten()
                    .into_iter()
                    .map(|c| ColumnCatalog {
                        column_desc: c,
                        is_hidden: col.is_hidden,
                    })
                    .collect_vec(),
        }
        self.columns = catalogs.clone();
        self
    }
```

This is a flattening process of a nested relation. So call it `flatten`.